client/webserver/site/src/js/doc.ts

```diff
@@ -258,15 +259,19 @@ export default class Doc {
     return decimalFormatter(prec).format(v)
   }
 
-  static formatThreeSigFigs (v: number): string {
-    if (v >= 1000) return intFormatter.format(Math.round(v))
-    return threeSigFigs.format(v)
+  /*
+   * formatRate formats rate to represent it exactly at rate step precision,
+   * trimming non-effectual zeros if there are any.
+   */
+  static formatRateToRateStep (encRate: number, bui: UnitInfo, qui: UnitInfo, rateStepEnc: number) {
```

Comment:
I've explored some tests to see if we're getting what we expect from this method. 

https://github.com/buck54321/dcrdex/compare/1a17cc4...buck54321:dcrdex:full-precision-rate-tests

Reply:
Nice, I was lazy to figure out how to run those, now I did,

I'm not sure how to use `npm run watch` properly (and it seems to be putting significant load on my laptop), so [I added `npm run build_dev`](https://github.com/decred/dcrdex/pull/2278/commits/cea84c2bb2c2b71282ff30c8e288e7737e226494) which is similar to `npm run build` but allows for accessing those tests ... if you don't have better suggestions for me on this.

Reply:
> I'm not sure how to use `npm run watch` properly

I think you have to use it with `--no-embed-site`, but I'm not sure either tbh.  It has been a while since I've done anything other than a full build.